apps/projects/app/components/Panel/NewProject/NewProject.js

```diff
@@ -63,160 +206,127 @@ const NewProject = () => {
-                  description: '',
-                  node: repo.node,
-                }))
-
-                return (
-                  <div>
-                    <TextInput
-                      type="search"
-                      style={{ margin: '16px 0', flexShrink: '0' }}
-                      placeholder="Search"
-                      wide
-                      value={filter}
-                      onChange={updateFilter}
-                      adornment={
-                        filter === '' && (
-                          <IconSearch
-                            css={`
-                                color: ${theme.surfaceOpened};
-                                margin-right: 8px;
-                              `}
-                          />
-                        )
-                      }
-                      adornmentPosition="end"
-                      ref={searchRef}
-                      aria-label="Search"
-                    />
+            <IconGitHub
+              color={`${theme.surfaceIcon}`}
+              width="18px"
+              height="18px"
+            />
+            <span css="margin-top: 3px; margin-left: 8px">Select from GitHub</span>
+          </Button>
+        </div>
+      ) : (
+        <RepoQuery
+          onRepoSelected={onRepoSelected}
+          repoSelected={repoSelected}
+          setRepoSelected={setRepoSelected}
+        />
+      )}
+      <Button
+        mode="strong"
+        wide
+        onClick={handleNewProject}
+        disabled={repoSelected < 0}
+      >
+        Submit
+      </Button>
+    </>
+  )
+}
+GitHubRepoList.propTypes = PropTypes.func.isRequired
 
-                    <ScrollableList>
-                      <RepoList visibleRepos={visibleRepos} repoArray={repoArray} />
-                    </ScrollableList>
-
-                    <Info css={`margin: ${3 * GU}px 0`}>
-                        Projects in Aragon are a one-to-one mapping to a GitHub repository.
-                        You’ll be able to prioritize your backlog, reach consensus on issue
-                        valuations, and allocate bounties to multiple issues.
-                    </Info>
-
-                    <Button
-                      mode="strong"
-                      wide
-                      onClick={handleNewProject}
-                      disabled={repoSelected < 0}
-                    >
-                      Submit
-                    </Button>
-                  </div>
-                )
+const createProject = () => {}
+
+const NewProject = ({ handleGithubSignIn }) => {
+  const [ title, setTitle ] = useState('')
+  const [ description, setDescription ] = useState('')
+  const theme = useTheme()
+  const { appState: { github = { status: STATUS.INITIAL } } } = useAragonApi()
+
+  return (
+    <PanelContent>
+      {
+        disableDecoupledProjects ? (
+          <GitHubRepoList handleGithubSignIn={handleGithubSignIn} />
+        ) : (
```

Comment:
an early return would look nicer, keep the indentation the way you had it, and make it easier to remove this feature-flipping later:

```jsx
if (disableDecoupledProjects) {
  return (
    <PanelContent>
      <GitHubRepoList handleGithubSignIn={handleGithubSignIn} />
    </PanelContent>
  )
}

return (
  <PanelContent>
    ...
  </PanelContent>
)
```

Reply:
Good point. Removing feature-flipping later will produce much less commit noise too. Changed.